// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.robotbas.devicesim";
option java_outer_classname = "DeviceSimProto";
option objc_class_prefix = "HLW";

package devicesim;

// The greeting service definition.
service DeviceService {
  // Sends a greeting
  // rpc SayHello (DeviceRequest) returns (HelloReply) {}

  rpc DeviceVariablesValues (DeviceRequest) returns (stream DeviceReply) {}

  // rpc SayHelloBidiStream (stream DeviceRequest) returns (stream HelloReply) {}
}

// The request message
message DeviceRequest {
}

// The response message containing the devices variables values
message DeviceReply {
  string device_name = 1;
  float temperature = 2;
  float humidity = 3;
  float pressure = 4;
  int64 timestamp = 5;
}
